server/mdm/apple/apple_mdm.go

```diff
@@ -256,6 +271,78 @@ func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team
 	return requestedTokenProfileUUID, requestedTokenModTime, nil
 }
 
+// ValidateSetupAssistant validates the setup assistant by sending the profile to the DefineProfile
+// Apple API.
+func (d *DEPService) ValidateSetupAssistant(ctx context.Context, team *fleet.Team, setupAsst *fleet.MDMAppleSetupAssistant, abmTokenOrgName string) error {
+	appCfg, err := d.ds.AppConfig(ctx)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "fetching app config")
+	}
+
+	// must always get the default profile, because the authentication token is
+	// defined on that profile.
+	defaultProf, err := d.ds.GetMDMAppleEnrollmentProfileByType(ctx, fleet.MDMAppleEnrollmentTypeAutomatic)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "fetching default profile")
+	}
+
+	enrollURL, err := EnrollURL(defaultProf.Token, appCfg)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "generating enroll URL")
+	}
+
+	rawJSON := setupAsst.Profile
+
+	jsonProf, err := d.buildJSONProfile(ctx, rawJSON, appCfg, team, enrollURL)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "building json profile")
+	}
+
+	depClient := NewDEPClient(d.depStorage, d.ds, d.logger)
+	// Get all relevant org names
+	var tmID *uint
+	if team != nil {
+		tmID = &team.ID
+	}
+
+	orgNames, err := d.ds.GetABMTokenOrgNamesAssociatedWithTeam(ctx, tmID)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "getting org names for team to register profile")
+	}
+
+	if len(orgNames) == 0 {
+		// Then check to see if there are any tokens at all. If there is only 1, we assume we can
+		// use it (the vast majority of deployments will only have a single token).
+		toks, err := d.ds.ListABMTokens(ctx)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "listing ABM tokens")
+		}
+
+		if len(toks) != 1 {
+			return ctxerr.New(ctx, "No relevant ABM tokens found. Please set this team as a default team for an ABM token.")
+		}
+
+		orgNames = append(orgNames, toks[0].OrganizationName)
+	}
+
+	for _, orgName := range orgNames {
+		_, err := depClient.DefineProfile(ctx, orgName, jsonProf)
+		if err != nil {
+			var httpErr *godep.HTTPError
+			if errors.As(err, &httpErr) {
+				// We can count on this working because of how the godep.HTTPerror Error() method
+				// formats its output.
+				parts := strings.Split(err.Error(), " ")
+				return ctxerr.Errorf(ctx, "Couldn't upload. %s", parts[len(parts)-1])
```

Comment:
I get it, but this is flaky, given that you already have asserted that's a `*godep.HTTPError`, why not use the `Body` attribute the error exposes?

```suggestion
				return ctxerr.Errorf(ctx, "Couldn't upload. %s", string(httpErr.Body))
```

Reply:
Oh! I'm not sure why, I thought there was other stuff in that `Body` field that we'd have to parse out anyway, but looks like I was wrong!